libraries/AP_RangeFinder/AP_RangeFinder_LightWareI2C.cpp

```diff
@@ -168,6 +169,14 @@ bool AP_RangeFinder_LightWareI2C::sf20_send_and_expect(const char* send_msg, con
     return memcmp(rx_bytes, expected_reply, expected_reply_len) == 0;
 }
 
+const char *AP_RangeFinder_LightWareI2C::get_version() const {
```

Comment:
Don't allocate any memory in this function. Instead have one argument for a char* buffer and another argument for the size of the buffer. Then use snprintf to write your version message.

Reply:
Let the buffer allocation be handled by the caller.

Reply:
I agree, this would reduce any complexity / chance for memory leak this way.

Reply:
Fixed!